website/static/js/contribManager.js

```diff
@@ -6,106 +6,131 @@ var bootbox = require('bootbox');
 require('jquery-ui');
 require('knockout-sortable');
 
+var rt = require('js/responsiveTable');
 var $osf = require('./osfHelpers');
 
-var contribsEqual = function(a, b) {
-    return a.id === b.id &&
-        a.visible === b.visible &&
-        a.permission === b.permission &&
-        Boolean(a.deleteStaged) === Boolean(b.deleteStaged);
-};
-
-// Modified from http://stackoverflow.com/questions/7837456/comparing-two-arrays-in-javascript
-var arraysEqual = function(a, b) {
-    var i = a.length;
-    if (i !== b.length) { return false;}
-    while (i--) {
-        if (!contribsEqual(a[i], b[i])) {return false;}
-    }
-    return true;
-};
-
-var sortMap = {
-    surname: {
-        label: 'Surname',
-        order: 1,
-        func: function(item) {
-            return item.surname;
-        }
-    }
+//http://stackoverflow.com/questions/12822954/get-previous-value-of-an-observable-in-subscribe-of-same-observable
+ko.subscribable.fn.subscribeChanged = function (callback) {
+    var self = this;
+    var savedValue = self.peek();
+    return self.subscribe(function (latestValue) {
+        var oldValue = savedValue;
+        savedValue = latestValue;
+        callback(latestValue, oldValue);
+    });
 };
 
 // TODO: We shouldn't need both pageOwner (the current user) and currentUserCanEdit. Separate
 // out the permissions-related functions and remove currentUserCanEdit.
-var ContributorModel = function(contributor, currentUserCanEdit, pageOwner, isRegistration, isAdmin) {
+var ContributorModel = function(contributor, currentUserCanEdit, pageOwner, isRegistration, isAdmin, index, parent) {
```

Comment:
Now that the parameter is list is large, it would be good to refactor this to take in a params object. This improves readability:

```js
new ContributorsModel(contribData, {canEdit: true, pageOwner: ..., isRegistration: false, });
```

But that doesn't necessarily need to happen before this is merged.